src/openrct2/object/RideObject.cpp

```diff
@@ -98,11 +98,13 @@ void RideObject::ReadLegacy(IReadObjectContext * context, IStream * stream)
         {
             numPeepLoadingPositions = stream->ReadValue<uint16>();
         }
-        _peepLoadingPositions[i] = stream->ReadArray<sint8>(numPeepLoadingPositions);
-        _peepLoadingPositionsCount[i] = numPeepLoadingPositions;
+
+        auto data = stream->ReadArray<sint8>(numPeepLoadingPositions);
+        _peepLoadingPositions[i] = std::vector<sint8>(data, data + numPeepLoadingPositions);
```

Comment:
Refer to https://github.com/OpenRCT2/OpenRCT2/pull/7225 there are two types of peep loading positions. I was planning on changing this so that it loads it differently depending on the type.